src/ui/components/inputs/Checkbox/Checkbox.tsx

```diff
@@ -1,84 +1,243 @@
 import { TouchableOpacity } from 'ui/components/TouchableOpacity'
 import { useSpaceBarAction } from 'ui/hooks/useSpaceBarAction'
-import { CheckboxMark } from 'ui/svg/icons/CheckBoxTMark'
-import { getSpacing, Typo } from 'ui/theme'
+import { CheckboxMarkChecked } from 'ui/svg/icons/CheckboxMarkChecked'
+import { CheckboxMarkIndeterminate } from 'ui/svg/icons/CheckboxMarkIndeterminate'
+import { Typo, getSpacing } from 'ui/theme'
+import { customFocusOutline } from 'ui/theme/customFocusOutline/customFocusOutline'
+import { getHoverStyle } from 'ui/theme/getHoverStyle/getHoverStyle'
 
-type IsCheckedProps = {
+type CheckboxBaseCheckedOnly = {
   isChecked: boolean
+  indeterminate?: false
 }
 
-type Props = IsCheckedProps & {
-  label?: string
-  LabelComponent?: FunctionComponent
+type CheckboxBaseIndeterminateOnly = {
+  isChecked: false
+  indeterminate: true
+}
+
+type CheckboxBase = {
+  label: string
   onPress: (isChecked: boolean) => void
   required?: boolean
-  style?: StyleProp<ViewStyle>
-}
+  hasError?: boolean
+  disabled?: boolean
+} & (CheckboxBaseCheckedOnly | CheckboxBaseIndeterminateOnly)
+
+type Props = CheckboxBase &
+  (
+    | {
+        asset?: never
+        collapsed?: never
+        description?: never
+        display?: CheckboxDisplay
+        variant?: 'default'
+      }
+    | {
+        asset?: CheckboxAssetProps
+        collapsed?: never
+        description?: string
+        display?: CheckboxDisplay
+        variant?: 'detailed'
+      }
+    | {
+        asset?: CheckboxAssetProps
+        collapsed: React.ReactNode
+        description?: string
+        display?: Extract<CheckboxDisplay, 'fill'>
+        variant?: 'detailed'
+      }
+  )
 
 export const Checkbox: FunctionComponent<Props> = ({
-  label,
+  asset,
+  collapsed,
+  description,
+  disabled,
+  display,
+  hasError,
+  indeterminate,
   isChecked,
-  required,
+  label,
   onPress,
-  LabelComponent = StyledBody,
-  style,
+  required,
+  variant = 'default',
 }) => {
-  const { onFocus, onBlur, isFocus } = useHandleFocus()
+  const focusProps = useHandleFocus()
+  const hoverProps = useHandleHover()
+
+  const { designSystem, checkbox } = useTheme()
 
   const onToggle = useCallback(() => {
     onPress(!isChecked)
   }, [isChecked, onPress])
 
-  useSpaceBarAction(isFocus ? onToggle : undefined)
+  useSpaceBarAction(focusProps.isFocus ? onToggle : undefined)
+
+  const effectiveDisplay: CheckboxDisplay = display ?? (variant === 'detailed' ? 'fill' : 'hug')
+
+  const state = getCheckboxState(isChecked, indeterminate, hasError, disabled)
+  const isDisabled = state.includes('disabled')
+  const isCheckedState = state.includes('checked')
+  const isIndeterminate = state.includes('indeterminate')
+
+  const colorMark = disabled ? designSystem.color.icon.disabled : designSystem.color.icon.inverted
+  const CheckboxMarkIndeterminateSize = checkbox.size / 1.75 // Parent padding doesn't have effect on CheckboxMarkIndeterminate
```

Comment:
détail : la case devrait etre en camel

```suggestion
  const checkboxMarkIndeterminateSize = checkbox.size / 1.75 // Parent padding doesn't have effect on CheckboxMarkIndeterminate
```